runtime/src/main/java/org/corfudb/protocols/wireprotocol/StateRes.java

```diff
@@ -0,0 +1,51 @@
+package org.corfudb.protocols.wireprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class StateRes implements ICorfuPayload<StateRes> {
+
+    @Getter
+    final Map<UUID, Long> tails;
+
+    @Getter
+    final long globalTail;
+
+    @Getter
+    final long epoch;
+
+
+    public StateRes(Map<UUID, Long> tails, long globalTail, long epoch) {
+        this.tails = tails;
+        this.globalTail = globalTail;
+        this.epoch = epoch;
+    }
+
+    public StateRes(ByteBuf buf) {
+        epoch = buf.readLong();
+        globalTail = buf.readLong();
+        int size = buf.readInt();
+        tails = new HashMap<>(size);
+        for (int x = 0; x < size; x++) {
+            UUID uuid = new UUID(buf.readLong(), buf.readLong());
+            long tail = buf.readLong();
+            tails.put(uuid, tail);
+        }
+    }
+
+    @Override
+    public void doSerialize(ByteBuf buf) {
+        buf.writeLong(epoch);
+        buf.writeLong(globalTail);
+        buf.writeInt(tails.size());
+        for (Map.Entry<UUID, Long> entry : tails.entrySet()) {
```

Comment:
You can use ICorfupayload.serialize to serialize all these variables instead of reimplementing this logic.

Reply:
Done.